trie/snap_diskcache.go

```diff
@@ -0,0 +1,204 @@
+// Copyright 2021 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package trie
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/VictoriaMetrics/fastcache"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var (
+	// cacheSizeLimit is the maximum size of the disk cache that aggregates
+	// the writes from above until it's flushed into the disk. Do not
+	// increase the cache size arbitrarily, otherwise the system pause
+	// time will increase when the database writes happen.
+	cacheSizeLimit = uint64(256 * 1024 * 1024)
```

Comment:
We should make this configurable IMHO (from the API). That would allow us to initialize in the constructor to one specific value (e.g. this constant), but then also have a method `SetCacheSize` that would allow us to lower it after we're in sync. That would allow us to play with large caches on startup and then lower it once in sync